src/Sparrow.Server/Allocators.Pool.cs

```diff
@@ -0,0 +1,312 @@
+﻿using System;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using Sparrow.Binary;
+using Sparrow.Global;
+
+namespace Sparrow.Server
+{
+    public interface IPoolAllocatorOptions : IAllocatorOptions, IComposableAllocator<Pointer>
+    {
+        int MaxBlockSize { get; }
+        int MaxPoolSizeInBytes { get; }
+    }
+
+    public static class PoolAllocator
+    {
+        public struct Default : IPoolAllocatorOptions, INativeOptions
+        {
+            public bool UseSecureMemory => false;
+            public bool ElectricFenceEnabled => false;
+            public bool Zeroed => false;
+
+            public int MaxBlockSize => 64 * Constants.Size.Megabyte;
```

Comment:
All these constants need to take into account 32 bits as well

Reply:
Until we have everything integrated those numbers are sensible defaults. After that we will need to looks at all different variants of allocators we end up using (I will create custom configurations for each) until we can then go there and find the actual 'default' case.